examples/src/migration/migration_to_aws_kms_keyring_from_aws_kms_master_key_provider_example.py

```diff
@@ -0,0 +1,272 @@
+
+For more information on how to use KMS keyrings, see
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/use-kms-keyring.html
+"""
+import boto3
+from aws_cryptographic_materialproviders.mpl import AwsCryptographicMaterialProviders
+from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+from aws_cryptographic_materialproviders.mpl.models import CreateAwsKmsKeyringInput
+from aws_cryptographic_materialproviders.mpl.references import IKeyring
+from typing import Dict  # noqa pylint: disable=wrong-import-order
+
+import aws_encryption_sdk
+
+EXAMPLE_DATA: bytes = b"Hello World"
+
+DEFAULT_ENCRYPTION_CONTEXT : Dict[str, str] = {
+    "encryption": "context",
+    "is not": "secret",
+    "but adds": "useful metadata",
+    "that can help you": "be confident that",
+    "the data you are handling": "is what you think it is",
+}
+
+
+class AwsKmsKeyring():
+    """Class for creating a KMS Keyring and using it for encryption and decryption"""
+
+    @staticmethod
+    def create_kms_client(aws_region="us-west-2"):
+        """Create an AWS KMS client.
+
+        Usage: create_kms_client(aws_region)
+        :param aws_region: AWS region to use for KMS client.
+        :type aws_region: string
+        """
+        # Create a boto3 client for KMS.
+        kms_client = boto3.client('kms', region_name=aws_region)
+
+        return kms_client
+
+    @staticmethod
+    def create_keyring(
+        kms_key_id: str
+    ):
+        """Demonstrate how to create an AWS KMS keyring.
+
+        Usage: create_keyring(kms_key_id)
+        :param kms_key_id: KMS Key identifier for the KMS key you want to use.
+        :type kms_key_id: string
+
+        For more information on KMS Key identifiers, see
+        https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id
+        """
+        # Create a boto3 client for KMS.
+        kms_client = AwsKmsKeyring.create_kms_client()
+
+        # Create a KMS keyring
+        mat_prov: AwsCryptographicMaterialProviders = AwsCryptographicMaterialProviders(
+            config=MaterialProvidersConfig()
+        )
+
+        keyring_input: CreateAwsKmsKeyringInput = CreateAwsKmsKeyringInput(
+            kms_key_id=kms_key_id,
+            kms_client=kms_client
+        )
+
+        keyring: IKeyring = mat_prov.create_aws_kms_keyring(
+            input=keyring_input
+        )
+
+        return keyring
+
+    @staticmethod
+    def encrypt_using_keyring(
+        plaintext_data: bytes,
+        keyring: IKeyring
+    ):
+        """Demonstrate how to encrypt plaintext data using an AWS KMS keyring.
+
+        Usage: encrypt_using_keyring(plaintext_data, keyring)
+        :param plaintext_data: plaintext data you want to encrypt
+        :type: bytes
+        :param keyring: Keyring to use for encryption.
+        :type keyring: IKeyring
+        """
+        client = aws_encryption_sdk.EncryptionSDKClient()
```

Comment:
Let's reuse the client across this module

Reply:
done